### PR TITLE
Golang versioned ovn-kube-util

### PR DESCRIPTION
**Pull request by feiskyer**
This PR adds golang versioned ovs-kube-util.

cc/ @shettyg

### CODE REVIEW

File: go-controller/cmd/ovn-kube-util/app/nics-to-bridge.go [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+}
+
+func nicToBridge(iface string) error {
+	ifaceLink, err := netlink.LinkByName(iface)
+	if err != nil {
+		return err
+	}
+
+	bridge := getBridgeName(iface)
+	stdout, stderr, err := util.RunOVSVsctl("add-br", bridge,
```
Comment (rajatchopra) [inline go-controller/cmd/ovn-kube-util/app/nics-to-bridge.go]: Is there an '--if-exists' flag to ovs-vsctl add-br? I cannot recall.

File: go-controller/cmd/ovn-kube-util/app/nics-to-bridge.go [hunk 2]
```diff
@@ -0,0 +1,134 @@
+package app
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"syscall"
+
+	"github.com/Sirupsen/logrus"
```
Comment (feiskyer) [inline go-controller/cmd/ovn-kube-util/app/nics-to-bridge.go]: I'd prefer logrus. WDYT?

Comment (rajatchopra) [inline go-controller/cmd/ovn-kube-util/app/nics-to-bridge.go]: logrus is just fine. Just seeking consistency. Pulling out glog in another PR then.

File: go-controller/Makefile [hunk 3]
```diff
@@ -7,6 +7,7 @@ export OUT_DIR
 #   make all
 all build:
 	hack/build-go.sh cmd/ovnkube/ovnkube.go
+	hack/build-go.sh cmd/ovn-kube-util/main.go
```
Comment (feiskyer) [inline go-controller/Makefile]: Yep, a unified binary with subcommands if preferred in the future.

File: go-controller/cmd/ovn-kube-util/app/nics-to-bridge.go (hunk 1 above)
Comment (feiskyer) [inline go-controller/cmd/ovn-kube-util/app/nics-to-bridge.go]: do you means `--may-exist`?

File: go-controller/cmd/ovn-kube-util/app/nics-to-bridge.go (hunk 2 above)
Comment (feiskyer) [inline go-controller/cmd/ovn-kube-util/app/nics-to-bridge.go]: ok. will do

File: go-controller/cmd/ovn-kube-util/app/nics-to-bridge.go [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+	Flags: []cli.Flag{},
+	Action: func(context *cli.Context) error {
+		args := context.Args()
+		if len(args) == 0 {
+			return fmt.Errorf("Please specify list of nic interfaces")
+		}
+
+		for _, nic := range args {
+			if err := nicToBridge(nic); err != nil {
+				return err
```
Comment (feiskyer) [inline go-controller/cmd/ovn-kube-util/app/nics-to-bridge.go]: ACK. Updated.

**Comment by shettyg** [comment]
This does not compile for me. I get the following error:

vendor/github.com/vishvananda/netlink/addr_linux.go:10:2: cannot find package "github.com/vishvananda/netlink/nl" in any of:

**Comment by feiskyer** [comment]
@shettyg Added back the missing vendors. PTAL.

**Comment by shettyg** [comment]
@feiskyer 
Thanks. It compiles now. But, looks like the broadcast address is not being properly moved to the bridge.

```
$tunctl -t p0
$ifconfig p0 192.168.50.2
$ifconfig p0
p0        Link encap:Ethernet  HWaddr 50:9a:1a:7d:21:1c  
          inet addr:192.168.50.2  Bcast:192.168.50.255  Mask:255.255.255.0
          UP BROADCAST MULTICAST  MTU:1500  Metric:1
          RX packets:0 errors:0 dropped:0 overruns:0 frame:0
          TX packets:0 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:500 
          RX bytes:0 (0.0 B)  TX bytes:0 (0.0 B)

$ovn-kube-util nics-to-bridge p0

$ifconfig brp0
brp0      Link encap:Ethernet  HWaddr 50:9a:1a:7d:21:1c  
          inet addr:192.168.50.2  Bcast:0.0.0.0  Mask:255.255.255.0
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:0 errors:0 dropped:0 overruns:0 frame:0
          TX packets:0 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:0 (0.0 B)  TX bytes:0 (0.0 B)
```

**Comment by feiskyer** [comment]
@shettyg Sorry, didn't notice this problem. It is because `vishvananda/netlink` always set broadcast to nil when getting addrs for ifaces. Filed a PR https://github.com/vishvananda/netlink/pull/226 to fix this.

**Comment by feiskyer** [comment]
Added a new commits with the fix. Should work properly now.

**Comment by shettyg** [comment]
@feiskyer 


LGTM

What is the typical strategy when to comes to vendor fixes in golang? Looks like here, you have updated our local vendor code. But, in the future if someone updates the vendor code, we will miss your fix, isn't it?

The other option is to have a local function that does the correct code and use it instead of the upstream vendor code. And then remove local code when the upstream has the fix. I am trying to understand what do other projects that use vendoring do?

**Comment by feiskyer** [comment]
We usually fix the problem outside vendors, but I'm expecting the fix could be merged soon in the upstream code, e.g. before this one.

**Comment by rajatchopra** [comment]
Same strategy in the openshift project. Fix with a specially marked commit and PR (e.g. keyword UPSTREAM) and actually rebase from all upstream when starting a new development release.

**Comment by feiskyer** [comment]
@shettyg  Added a local version `addrList` for a workaround.

**Pull request merged**